virtcontainers/kata_agent.go

```diff
@@ -625,13 +625,17 @@ func constraintGRPCSpec(grpcSpec *grpc.Spec) {
 
 func (k *kataAgent) appendDevices(deviceList []*grpc.Device, devices []api.Device) []*grpc.Device {
 	for _, device := range devices {
-		d, ok := device.(*drivers.BlockDevice)
-		if !ok {
+		if device.DeviceType() != config.DeviceBlock {
+			continue
+		}
+		d, ok := device.GetDeviceDrive().(*config.BlockDrive)
+		if !ok || d == nil {
+			k.Logger().Error("malformed block drive")
```

Comment:
I'd add details of the invalid device. Something like:

```
k.Logger().WithField("device", device).Error("malformed block drive")
```

Same comment to error below.